plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcClient.java

```diff
@@ -154,6 +154,11 @@ default void setTableProperties(ConnectorSession session, JdbcTableHandle handle
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting table properties");
     }
 
+    default ConnectorTableMetadata getNewTableMetadata(ConnectorSession connectorSession, SchemaTableName schemaTableName, List<ColumnMetadata> columns, Map<String, Object> properties, Optional<String> comment)
```

Comment:
JdbcClient interface should not need to operate on `ConnectorTableMetadata`. It should be granular, on per type basis.

Actually, we'd want to _change_ or _replace_ existing operation: `io.trino.plugin.jdbc.JdbcClient#toWriteMapping`. 
The JdbcClient would want to return replacement `Type` that would go into the effective new table metadata.